### ISSUE TITLE
Allow OIDC usage over http

### ISSUE THREAD

**Issue by KnechtionsCoding** [elastic/elasticsearch#88541]
### Background

Elasticsearch requires TLS connections in order to enable some security feature like [Tokens for OIDC](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html#security-api-create-api-key-desc).
This is important to guarantee that sensitive information is properly protected.

However, there are scenarios where encryption is managed outside of Elasticsearch.
Notable examples are [service mesh](https://en.wikipedia.org/wiki/Service_mesh) or [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) software that are in front of the cluster, handling all incoming communication with the REST endpoint.

#### Goal
Allow service mesh and reverse proxy environments to use Elasticsearch OIDC/Tokens in scenarios where a clear-text connection between the service and Elasticsearch is used.

#### Proposal
Allow to relax the TLS bootstrap checks to make OIDC/Tokens available when TLS is not enabled. 

#### Other background Discussion:

https://github.com/dexidp/dex/issues/1593

https://github.com/elastic/elasticsearch/issues/61458

**Comment by ywangd** [comment]
This is a duplicate of https://github.com/elastic/elasticsearch/issues/81817. But I am going to close #81817 since this issue here is just for OAuth2 tokens while the other one has a mix of tokens and API keys and API keys no longer have the TLS restriction.

**Comment by danielabbatt** [comment]
Yes this would be very useful - I can't currently use Entra via OpenId Connect where the cluster is in kubernetes behind an haproxy which is TLS terminated. The pods inside the cluster are running just http so we don't have to deal with assigning certs for termination.

**Comment by tomas-pritrsky** [comment]
This is an issue.

**Comment by landerz123** [comment]
Is there any progress on that? @elasticsearchmachine

**Comment by Anankke** [comment]
Any updates on this?

**Comment by jtele2** [comment]
+1

**Comment by spidaparthi** [comment]
Any update on this, we have similar issue when trying to setup SAML on GKE with Istio STRICT mTLS.

**Comment by aviau** [comment]
We would love to see this.

**Comment by jtele2** [comment]
Any progress on this? With service meshes being so common now, we'd really like to see this implemented.

**Comment by jada-mak** [comment]
Any progress on this? We're having the same issue with Istio.

**Comment by dklenke** [comment]
We would love to see this feature as well
